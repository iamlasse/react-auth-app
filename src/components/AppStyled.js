import styled from 'styled-components'
import { NavLink } from 'react-router-dom'
import { Colors } from '@blueprintjs/core'

export const StyledWrapper = styled.div`
			display: flex
			flex: 1
			align-items: center
			justify-content: center
			height: 100vh
			width: 100vw
			text-align: center
			flex-direction: column
`

export const StyledWrappedMain = styled.div`
    display: flex
    justify-content: center
    align-items: center
    height: 100vh
    width: 100vw
    background-color: #f9f9f9
`

export const StyledLink = styled(NavLink)`
	
	align-items: center;
    justify-content: center;
		display: flex;
		height: 100%

	padding: 10px 20px
	font-weight: bold
	transition: all 0.5s ease
	&:hover {
		text-decoration: none;
			
	}
	&.active {
		outline: none;
		color: ${Colors.WHITE}
		background-color: ${Colors.BLUE3}
	}
`
