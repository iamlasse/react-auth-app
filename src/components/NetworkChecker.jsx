import React, { Component } from 'react'
import PropTypes from 'prop-types'
import AppToaster from "../constants/AppToaster";
// import Toast from "../Toast";

export default class NetworkChecker extends Component {

  static defaultProps = {
    online: "Online",
    offline: "Offline"
  }

  static propTypes = {
    online: PropTypes.string,
    offline: PropTypes.string,
  }

  componentDidMount() {
    window.addEventListener("offline", this.handleConnectionStatus)
    window.addEventListener("online", this.handleConnectionStatus)
    if (!window.navigator.onLine) {
      this.handleConnectionStatus()
    }
  }

  componentWillUnmount = () => {
    window.removeEventListener("offline", this.handleConnectionStatus)
    window.removeEventListener("online", this.handleConnectionStatus)
  }

  handleConnectionStatus = (event) => {
    const { online, offline } = this.props
    const condition = window.navigator.onLine ? "online" : "offline";
    const message = condition === 'online' ? online : offline
    const intent = condition === 'online' ? 'success' : 'danger'
    console.log(`Network is: ${condition}`, event && event.type);
    AppToaster.show({
      message,
      intent
    })
  }

  render() {
    return (
      null
    )
  }
}
