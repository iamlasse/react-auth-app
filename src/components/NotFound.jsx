import React from 'react'
import { Image, Box, Heading } from 'grommet'
import { Link } from 'react-router-dom'

import notFoundImage from '../assets/images/not-found.png'

const NotFound = () => (
	<Box pad="small" gap="small">
		<Box width="large" height="large" align="center" justify="center">
			<Image src={notFoundImage} fit="contain" style={{ width: 500 }} />
			<Heading level={1} size="xlarge" margin="xsmall" style={{ marginTop: -60 }}>
				404!
			</Heading>
			<Heading level={2}>Error, This page doesn't exist</Heading>
			<Link to="/">Go to Homepage</Link>
		</Box>
	</Box>
)

export default NotFound
