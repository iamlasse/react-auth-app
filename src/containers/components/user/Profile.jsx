import React from 'react';
import PropTypes from 'prop-types';
import { Card, Elevation, H3 } from '@blueprintjs/core';

const Profile = props => {
	const { profile } = props;
	return (
		<div>
			<Card elevation={Elevation.ONE}>
				<H3>{profile.username}</H3>
				<p>{profile.email}</p>
			</Card>
		</div>
	);
};

Profile.propTypes = {
	profile: PropTypes.instanceOf(Object)
};

Profile.defaultProps = {
	profile: {}
};

export default Profile;
