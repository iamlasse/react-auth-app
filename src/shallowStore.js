import React from 'react';
import { shallow } from 'enzyme';
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

const shallowWithStore = (component, store) => {
	const context = {
		store
	};
	return shallow(<component />, { context });
};

export default shallowWithStore;
