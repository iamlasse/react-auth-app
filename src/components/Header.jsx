import React from 'react'
import PropTypes from 'prop-types'
import { Navbar, Alignment, Classes, Icon } from '@blueprintjs/core'
import { Box } from 'grommet'
import { StyledLink } from './AppStyled'
import UserMenu from './UserMenu'

const Header = ({ auth: { authenticated }, logout, theme }) => {
	const isHome = (match, location) => {
		if (location.pathname !== '/') return false
		return true
	}

	return (
		<Box style={{ position: 'fixed', width: '100vw' }}>
			<Navbar className={theme === 'dark' ? Classes.DARK : ''}>
				<Navbar.Group align={Alignment.LEFT}>
					<StyledLink to="/" isActive={isHome}>
						<Icon icon="home" style={{ marginRight: 5 }} /> Home
					</StyledLink>
					{authenticated && (
						<StyledLink to="/dashboard" style={{ textDecoration: 'none' }}>
							<Icon icon="dashboard" style={{ marginRight: 5 }} />
							Dashboard
						</StyledLink>
					)}
				</Navbar.Group>
				<UserMenu authenticated={authenticated} handleLogout={logout} />
			</Navbar>
		</Box>
	)
}

Header.propTypes = {
	auth: PropTypes.instanceOf(Object),
	authenticated: PropTypes.bool,
	logout: PropTypes.func,
	theme: PropTypes.instanceOf(Object)
}

Header.defaultProps = {
	auth: {},
	authenticated: null,
	logout: () => null,
	theme: {}
}

export default Header
