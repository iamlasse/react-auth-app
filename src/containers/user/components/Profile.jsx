import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import { Route, Link } from 'react-router-dom'
import { Card, Elevation, H3, Spinner, Button, Switch } from '@blueprintjs/core'
import { StyledWrapper } from '../../../components/AppStyled'
import AddNote from './AddNote'

class Profile extends React.Component {
	state = {
		darkTheme: true
	}

	componentDidMount() {
		const { getNotesAsync, theme } = this.props
		getNotesAsync()
		this.setState({ darkTheme: theme === 'dark' })
	}

	componentWillReceiveProps(nextProps) {
		const { notes } = this.props
		if (nextProps.notes.length !== notes.length) {
			const { getNotesAsync } = this.props
			getNotesAsync()
		}
	}

	renderNotes = url => {
		const { notes, selectNote } = this.props
		return (
			<ul>
				{notes.map(note => (
					<li key={note.id}>
						<Link to={`${url}/note/${note.id}`} onClick={() => selectNote(note.id)}>
							{' '}
							{note.title}
						</Link>
					</li>
				))}
				<li>
					<Link to={`${url}/addNote`}> Add Note</Link>
				</li>
			</ul>
		)
	}

	renderNote = note => (
		<div>
			<h4> {note.title}</h4> <p> {note.text}</p>
		</div>
	)

	saveSettings = () => {
		const { saveSettingsAsync } = this.props
		const { darkTheme } = this.state
		const settings = {
			theme: darkTheme ? 'dark' : 'light',
			lastLogin: new Date()
		}
		saveSettingsAsync(settings)
	}

	handleThemeChange = () => {
		const { darkTheme } = this.state
		this.setState({ darkTheme: !darkTheme })
	}

	render() {
		const {
			profile,
			notes,
			match: { url },
			fetching,
			selectedNote
		} = this.props
		const { darkTheme } = this.state
		return (
			<StyledWrapper>
				<Card elevation={Elevation.ONE}>
					<H3> {profile.username}</H3> <p> {profile.email}</p>
					<Switch
						checked={darkTheme}
						label="Dark Theme"
						onChange={this.handleThemeChange}
					/>
					<Button onClick={this.saveSettings}>Save Settings</Button>
				</Card>{' '}
				{fetching && <Spinner />}{' '}
				{!fetching && (
					<div className="notes">
						{' '}
						{notes && this.renderNotes(url)}{' '}
						{!notes && (
							<h3>
								NO Notes... <Link to={`${url}/addNote`}> add one </Link>
							</h3>
						)}
					</div>
				)}
				<Route exact path={`${url}/addNote`} render={() => <AddNote />} />{' '}
				<Route
					path={`${url}/note/:noteId`}
					render={() => {
						const note = selectedNote
						return <Fragment>{this.renderNote(note)}</Fragment>
					}}
				/>
			</StyledWrapper>
		)
	}
}

Profile.propTypes = {
	profile: PropTypes.instanceOf(Object),
	theme: PropTypes.instanceOf(Object),
	notes: PropTypes.instanceOf(Object),
	getNotesAsync: PropTypes.func,
	fetching: PropTypes.bool,
	selectNote: PropTypes.func,
	selectedNote: PropTypes.instanceOf(Object),
	match: PropTypes.instanceOf(Object),
	saveSettingsAsync: PropTypes.func
}

Profile.defaultProps = {
	profile: {},
	theme: {},
	notes: [],
	getNotesAsync: null,
	fetching: false,
	selectNote: null,
	selectedNote: {},
	match: {},
	saveSettingsAsync: null
}

export default Profile
