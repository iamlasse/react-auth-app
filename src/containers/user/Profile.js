import { connect } from 'react-redux';
import Profile from '../components/user/Profile';
import { authSel } from '../../store/modules/auth';

const { getAuthUser } = authSel;

const mapStateToProps = state => ({
	profile: getAuthUser(state)
});

export default connect(mapStateToProps)(Profile);
