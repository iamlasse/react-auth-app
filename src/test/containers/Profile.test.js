import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import { compose } from 'redux';
import { withRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import { connect } from 'react-redux';
import * as shallowWithStore from '../shallowStore';
import { shallow } from 'enzyme';
import * as Profile from '../../components/Profile';

const middlewares = [];
const mockStore = configureMockStore(middlewares);

let Component;
let wrapper;
describe('Profile Component', () => {
	beforeEach(() => {
		Component = compose(withRouter(connect(() => {})(Profile)));
		// wrapper = shallow(<Component />);
	});

	it('renders without crashing', () => {
		const testState = {
			user: {}
		};

		const initialState = {};
		const store = mockStore(testState);
		// Component = compose(Profile);
		const component = shallowWithStore(<Component />, store);
		expect(component).to.be.a('object');
	});

	// it('has propTypes', () => {
	// 	expect(Profile.propTypes).objectContaining({ profile: PropTypes.instanceOf(Object) });
	// });
});
