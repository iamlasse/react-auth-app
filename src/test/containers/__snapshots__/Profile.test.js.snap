// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component renders without crashing 1`] = `undefined`;
